New MissingPermission warnings with AGP 8.10

Changes to the Android linter currently in the alpha stage make the MissingPermission checks stricter than they currently are. As a result, reference-browser (and Fenix) now trigger warnings where they previously didn't:
```
  /app/src/main/java/org/mozilla/reference/browser/NotificationManager.kt:99: Error: Call requires permission which may be rejected by user: code should explicitly check to see if permission is available (with checkPermission) or explicitly handle a potential SecurityException [MissingPermission]
              NotificationManagerCompat.from(context).notify(
              ^
  /app/src/main/java/org/mozilla/reference/browser/NotificationManager.kt:152: Error: Call requires permission which may be rejected by user: code should explicitly check to see if permission is available (with checkPermission) or explicitly handle a potential SecurityException [MissingPermission]
          NotificationManagerCompat.from(context)
          ^
  
     Explanation for issues of type "MissingPermission":
     This check scans through your code and libraries and looks at the APIs
     being used, and checks this against the set of permissions required to
     access those APIs. If the code using those APIs is called at runtime, then
     the program will crash.
  
     Furthermore, for permissions that are revocable (with targetSdkVersion 23),
     client code must also be prepared to handle the calls throwing an exception
     if the user rejects the request for permission at runtime.
```

--- a/app/src/main/java/org/mozilla/reference/browser/NotificationManager.kt
+++ b/app/src/main/java/org/mozilla/reference/browser/NotificationManager.kt
@@ -4,6 +4,7 @@
 
 package org.mozilla.reference.browser
 
+import android.annotation.SuppressLint
 import android.app.Notification
 import android.app.NotificationChannel
 import android.app.PendingIntent
@@ -48,6 +49,14 @@ object NotificationManager {
 
     private val logger = Logger("NotificationManager")
 
+    /**
+     * Displays a notification for each valid tab received from a device.
+     *
+     * @param context The Android application context.
+     * @param device The device from which the tabs were received, or `null` if unavailable.
+     * @param tabs A list of tabs to be displayed.
+     */
+    @SuppressLint("MissingPermission", "NotifyUsage")
     fun showReceivedTabs(context: Context, device: Device?, tabs: List<TabData>) {
         // In the future, experiment with displaying multiple tabs from the same device as as Notification Groups.
         // For now, a single notification per tab received will suffice.
@@ -117,6 +126,7 @@ object NotificationManager {
     /**
      * Launch a notification of the data policy, and record notification time and version.
      */
+    @SuppressLint("MissingPermission", "NotifyUsage")
     private fun notifyDataPolicy(context: Context, preferences: SharedPreferences) {
         val resources = context.resources
 

--- a/config/detekt-baseline.xml
+++ b/config/detekt-baseline.xml
@@ -80,7 +80,6 @@
     <ID>UndocumentedPublicFunction:ExternalAppBrowserFragment.kt$ExternalAppBrowserFragment.Companion$fun create( sessionId: String, manifest: WebAppManifest?, trustedScopes: List&lt;Uri&gt;, )</ID>
     <ID>UndocumentedPublicFunction:Long.kt$@Suppress("MagicNumber") fun Long.timeSince(context: Context, time: Long): String</ID>
     <ID>UndocumentedPublicFunction:NotificationManager.kt$NotificationManager$fun checkAndNotifyPolicy(context: Context)</ID>
-    <ID>UndocumentedPublicFunction:NotificationManager.kt$NotificationManager$fun showReceivedTabs(context: Context, device: Device?, tabs: List&lt;TabData&gt;)</ID>
     <ID>UndocumentedPublicFunction:PictureInPictureIntegration.kt$PictureInPictureIntegration$fun onHomePressed()</ID>
     <ID>UndocumentedPublicFunction:Settings.kt$Settings$fun getOverrideAmoCollection(context: Context): String</ID>
     <ID>UndocumentedPublicFunction:Settings.kt$Settings$fun getOverrideAmoUser(context: Context): String</ID>

